src/jit/codegenxarch.cpp

```diff
@@ -2470,9 +2470,21 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
     case GT_JTRUE:
         {
             GenTree *cmp = treeNode->gtOp.gtOp1;
+
             assert(cmp->OperIsCompare());
             assert(compiler->compCurBB->bbJumpKind == BBJ_COND);
 
+#if !defined(_TARGET_64BIT_)
+            GenTreePtr op1 = cmp->gtGetOp1();
+
+            // For long compares, we emit special logic
+            if (varTypeIsLong(op1->TypeGet()))
```

Comment:
(Minor) If you would prefer we can avoid declaring op1 local and calling TypeGet() by using 

`if (varTypeIsLong(cmp->gtGetOp1()))`

Note that varTypeIsLong() is a template method that also accepts GenTreePtr as argument.